userspace/libpman/src/maps.c

```diff
@@ -75,76 +75,159 @@ uint64_t pman_get_probe_schema_ver() {
-	g_state.skel->bss->g_settings.sampling_ratio = value;
+	struct capture_settings settings;
+	pman_get_capture_settings(&settings);
+	settings.sampling_ratio = value;
+	pman_update_capture_settings(&settings);
 }
 
 void pman_set_drop_failed(bool drop_failed) {
-	g_state.skel->bss->g_settings.drop_failed = drop_failed;
+	struct capture_settings settings;
+	pman_get_capture_settings(&settings);
+	settings.drop_failed = drop_failed;
+	pman_update_capture_settings(&settings);
 }
 
 void pman_set_do_dynamic_snaplen(bool do_dynamic_snaplen) {
-	g_state.skel->bss->g_settings.do_dynamic_snaplen = do_dynamic_snaplen;
+	struct capture_settings settings;
+	pman_get_capture_settings(&settings);
+	settings.do_dynamic_snaplen = do_dynamic_snaplen;
+	pman_update_capture_settings(&settings);
 }
 
 void pman_set_fullcapture_port_range(uint16_t range_start, uint16_t range_end) {
-	g_state.skel->bss->g_settings.fullcapture_port_range_start = range_start;
-	g_state.skel->bss->g_settings.fullcapture_port_range_end = range_end;
+	struct capture_settings settings;
+	pman_get_capture_settings(&settings);
+	settings.fullcapture_port_range_start = range_start;
+	settings.fullcapture_port_range_end = range_end;
+	pman_update_capture_settings(&settings);
 }
 
 void pman_set_statsd_port(uint16_t statsd_port) {
-	g_state.skel->bss->g_settings.statsd_port = statsd_port;
+	struct capture_settings settings;
+	pman_get_capture_settings(&settings);
+	settings.statsd_port = statsd_port;
+	pman_update_capture_settings(&settings);
 }
 
 void pman_set_scap_tid(int32_t scap_tid) {
-	g_state.skel->bss->g_settings.scap_tid = scap_tid;
-}
-
-void pman_mark_single_64bit_syscall(int intersting_syscall_id, bool interesting) {
-	g_state.skel->bss->g_64bit_interesting_syscalls_table[intersting_syscall_id] = interesting;
+	struct capture_settings settings;
+	pman_get_capture_settings(&settings);
+	settings.scap_tid = scap_tid;
+	pman_update_capture_settings(&settings);
 }
 
 void pman_fill_syscall_sampling_table() {
 	for(int syscall_id = 0; syscall_id < SYSCALL_TABLE_SIZE; syscall_id++) {
 		if(g_syscall_table[syscall_id].flags & UF_NEVER_DROP) {
-			g_state.skel->bss->g_64bit_sampling_syscall_table[syscall_id] = UF_NEVER_DROP;
+			g_state.skel->rodata->g_64bit_sampling_syscall_table[syscall_id] = UF_NEVER_DROP;
 			continue;
 		}
 
 		/* Syscalls with `g_syscall_table[syscall_id].flags == UF_NONE` are the generic ones */
 		if(g_syscall_table[syscall_id].flags & UF_ALWAYS_DROP ||
 		   g_syscall_table[syscall_id].flags == UF_NONE) {
-			g_state.skel->bss->g_64bit_sampling_syscall_table[syscall_id] = UF_ALWAYS_DROP;
+			g_state.skel->rodata->g_64bit_sampling_syscall_table[syscall_id] = UF_ALWAYS_DROP;
 			continue;
 		}
 
 		if(g_syscall_table[syscall_id].flags & UF_USED) {
-			g_state.skel->bss->g_64bit_sampling_syscall_table[syscall_id] = 0;
+			g_state.skel->rodata->g_64bit_sampling_syscall_table[syscall_id] = 0;
 			continue;
 		}
 	}
 }
 
+int pman_init_settings_map() {
```

Comment:
it sounds like `pman_update_capture_settings` with an initial `struct capture_settings settings = {};` maybe we can avoid duplication

Reply:
Good catch, implemented your suggestion